profile_service/src/main/kotlin/ru/zveron/mapper/ContactsMapper.kt

```diff
@@ -19,17 +21,70 @@ object ContactsMapper {
             chat = true
         )
 
-    fun Contact.toModel(): Links =
+    fun Links.toCommunicationLinks(profile: Profile): List<CommunicationLink> {
+        val result = mutableListOf<CommunicationLink>()
+        if (vk.id.isNotBlank()) {
+            result.add(
+                CommunicationLink(
+                    communicationLinkId = vk.id,
+                    data = VkData(
+                        ref = vk.ref,
+                        email = vk.email,
+                    ),
+                    profile = profile
+                )
+            )
+        }
+        if (gmail.id.isNotBlank()) {
+            result.add(
+                CommunicationLink(
+                    communicationLinkId = gmail.id,
+                    data = GmailData(
+                        email = gmail.email,
+                    ),
+                    profile = profile
+                )
+            )
+        }
+        if (phone.number.isNotBlank()) {
+            result.add(
+                CommunicationLink(
+                    communicationLinkId = phone.number,
+                    data = PhoneData(),
+                    profile = profile
+                )
+            )
+        }
+        return result
+    }
```

Comment:
```suggestion
fun Links.toCommunicationLinks(profile: Profile): List<CommunicationLink> = listOfNotNull(
        vk.takeIf { it.id.isNotEmpty() }?.let {
            CommunicationLink(
                communicationLinkId = it.id,
                data = VkData(
                    ref = it.ref,
                    email = it.email,
                ),
                profile = profile
            )
        },
        gmail.takeIf { it.id.isNotEmpty() }?.let {
            CommunicationLink(
                communicationLinkId = it.id,
                data = GmailData(
                    email = it.email,
                ),
                profile = profile
            )
        },
        phone.takeIf { it.number.isNotEmpty() }?.let {
            CommunicationLink(
                communicationLinkId = phone.number,
                data = PhoneData(),
                profile = profile
            )
        },
    )
```
минор и вопрос вкусовщины, но я бы так сделал

Reply:
еще takeIf { isNotEmpty } можно в экстеншон затолкать и называть takeIfNotEmpty